dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+	            context.put("friendlyName", htmlPage.getFriendlyName());
+	            context.put("pageTitle", htmlPage.getTitle());
+                context.put("dotPageMode", mode);
+	            Date moddate = htmlPage.getModDate();
+
+	            moddate = new Date(moddate.getTime());
+
+	            context.put("HTML_PAGE_LAST_MOD_DATE", moddate);
+
+	            try {
+	                context.put("htmlPageInode", htmlPage.getInode());
+
+	                // for browsing the tree
+	                String view = java.net.URLEncoder.encode("(working=" + com.dotmarketing.db.DbConnectionFactory.getDBTrue()
+	                        + " and deleted=" + com.dotmarketing.db.DbConnectionFactory.getDBFalse() + "and language_id = "
+	                        + (String) request.getSession().getAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE) + ")", "UTF-8");
+	                context.put("view", view);
+	            } catch (Exception e) {
+	                Logger.warn(VelocityUtil.class, e.toString(), e);
+	            }
+	        }
+
+	        context.put("HTMLPAGE_SERVER_NAME", request.getServerName());
+	        context.put("VTLSERVLET_URI", UtilMethods.encodeURIComponent(idURI));
+	        if (request.getQueryString() != null && request.getQueryString().length() > 0) {
+	            context.put("queryString", request.getQueryString());
+	        } else {
+	            context.put("queryString", "");
+	        }
+	        context.put("TEMPLATE_INODE", String.valueOf(cmsTemplateInode));
+
+	        context.put("mainFrame", request.getParameter("mainFrame"));
+	        context.put("previewFrame", request.getParameter("previewFrame"));
+	        context.put(mode.name(), true);
+	        if (mode == PageMode.EDIT_MODE) {
+	            // gets user id from request for mod user
+	            com.liferay.portal.model.User backendUser = null;
+
+	            try {
+	                backendUser = com.liferay.portal.util.PortalUtil.getUser(request);
+	                // Skin skin = backendUser.getSkin();
+	                // context.put("USER_SKIN", skin.getSkinId());
+	            } catch (Exception nsue) {
+	                Logger.warn(VelocityUtil.class, "Exception trying yo getUser: " + nsue.getMessage(), nsue);
+	            }
+
+	            // to check user has permission to publish this page
+	            boolean permission = APILocator.getPermissionAPI().doesUserHavePermission(htmlPage, PERMISSION_PUBLISH, backendUser);
+	            context.put("permission", new Boolean(permission));
+
+	            // Check if the user is a CMS Administrator
+	            boolean adminUser = false;
+	            try {
+
+
+	                String adminRoleKey = "";
+	                try {
+	                    Role adminRole = APILocator.getRoleAPI().loadRoleByKey(Config.getStringProperty("CMS_ADMINISTRATOR_ROLE", "CMS Administrator"));
+	                    adminRoleKey = adminRole.getRoleKey();
+	                } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00108)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)